.github/workflows/maven.yml

```diff
@@ -11,10 +11,10 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
```

Comment:
this needs an if too, to switch between ubuntu and windows

Reply:
https://github.com/aws-greengrass/aws-greengrass-nucleus/commit/eeaa4cbf6dc934d641c97449ce05027b768e0e9b#diff-5dbf1a803ecc13ff945a08ed3eb09149a83615e83f15320550af8e3a91976446L14
this is how we used to do it.

Reply:
I am testing against `ubuntu-latest` and `windows-latest` within the steps. Would that be OK with you? Hope this is what you mean.

Reply:
I guess so, whatever works